### ISSUE TITLE
Telepathy support (includes Empathy)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dequis/msn-pecan#98]
```
Telepathy in general.


```

Original issue reported on code.google.com by `felipe.contreras` on 27 Sep 2008 at 2:58

**Comment by GoogleCodeExporter** [comment]
Original comment by `felipe.contreras` on 7 Mar 2009 at 9:58
- Changed title: **Telepathy support**

**Comment by GoogleCodeExporter** [comment]
```
telepathy?  seriously?  
```

Original comment by `samoore...@gmail.com` on 11 Jun 2009 at 10:29

**Comment by GoogleCodeExporter** [comment]
```
Yeah, distributions are seriously considering to drop Pidgin and use Empathy by 
default, it's now part of GNOME, and it's also used on the Nokia Internet 
Tablets.

Not to mention that Telepathy has had voice and video support since a long time.
```

Original comment by `felipe.contreras` on 11 Jun 2009 at 10:34

**Comment by GoogleCodeExporter** [comment]
Original comment by `felipe.contreras` on 20 Jun 2009 at 10:58
- Changed state: **Started**
- Added labels: **Milestone-1.0.0**

**Comment by GoogleCodeExporter** [comment]
```
They might be considering it, but pidgin won't be dropped till empathy stops 
sucking
```

Original comment by `matthaeu...@gmail.com` on 6 Jul 2009 at 5:19

**Comment by GoogleCodeExporter** [comment]
```
Actually, they all have in their plans for the next release; Fedora, Ubuntu and
OpenSUSE. So most probably Pidgin will be dropped.

Also Empathy supports audio/video already.
```

Original comment by `felipe.contreras` on 6 Jul 2009 at 6:00

**Comment by GoogleCodeExporter** [comment]
```
Oh yes, I prefer pidgin but if can be a pecan for telepaty I will try for sure. 
The
first problem for me are BBcode of my contact list.
```

Original comment by `mastrop...@gmail.com` on 22 Jul 2009 at 7:02

**Comment by GoogleCodeExporter** [comment]
```
Huh? BBcode is displayed correctly in msn-pecan, right? It would show the same 
in
Pidgin and Empathy.
```

Original comment by `felipe.contreras` on 22 Jul 2009 at 12:56

**Comment by GoogleCodeExporter** [comment]
```
Well, i was wrong. Ubuntu Karmic has officially dropped Pidgin and added 
Empathy.
```

Original comment by `matthaeu...@gmail.com` on 22 Jul 2009 at 5:45

**Comment by GoogleCodeExporter** [comment]
```
@matthaeus123 for now, yes, so has Fedora and OpenSUSE. If all goes well 
they'll be
shipping it in the next version. If not, perhaps the next one, but I think it's 
fair
to say that no one sees future in Pidgin.
```

Original comment by `felipe.contreras` on 22 Jul 2009 at 8:42

**Comment by GoogleCodeExporter** [comment]
Original comment by `felipe.contreras` on 23 Jul 2009 at 6:47
- Changed title: **Telepathy support (includes Empathy)**

**Comment by GoogleCodeExporter** [comment]
```
Issue 188 has been merged into this issue.
```

Original comment by `felipe.contreras` on 23 Jul 2009 at 6:51

**Comment by GoogleCodeExporter** [comment]
```
Does msn-pecan work with telepathy-haze?
```

Original comment by `devid...@gmail.com` on 23 Jul 2009 at 10:50

**Comment by GoogleCodeExporter** [comment]
```
@devidfil it should require minor modifications to telepathy-haze, but yeah.
```

Original comment by `felipe.contreras` on 23 Jul 2009 at 11:24

**Comment by GoogleCodeExporter** [comment]
```
AFAIK in Ubuntu Karmic telepathy-haze will be used, so I think for the moment 
we should work on that..
```

Original comment by `devid...@gmail.com` on 24 Jul 2009 at 7:56

**Comment by GoogleCodeExporter** [comment]
```
@ felipe, on pidgin is all ok; on empathy I can't find the download link, sorry 
my
fault, can u link me it?
```

Original comment by `mastrop...@gmail.com` on 26 Jul 2009 at 1:44

**Comment by GoogleCodeExporter** [comment]
```
I download from git, but I take this error:

gcc: libmsn-pecan/libmsn-pecan.a: Nessun file o directory

I downloaded libmsn-pecan from git and I put it into telepathy-msn-pecan dir 
but the
problem persist.
```

Original comment by `mastrop...@gmail.com` on 26 Jul 2009 at 2:39

**Comment by GoogleCodeExporter** [comment]
```
@mastropino msn-pecan doesn't work on Telepathy yet, that's why this ticket is 
open.
```

Original comment by `felipe.contreras` on 26 Jul 2009 at 5:49

**Comment by GoogleCodeExporter** [comment]
```
Please test http://github.com/dfiloni/msn-pecan/commits/telepathy (telepathy 
branch).
```

Original comment by `devid...@gmail.com` on 29 Aug 2009 at 10:07

**Comment by GoogleCodeExporter** [comment]
```
I gave it a go but I'm just getting 'network error'.
```

Original comment by `wyattmei...@gmail.com` on 29 Aug 2009 at 11:30

**Comment by GoogleCodeExporter** [comment]
```
Can you attach a debug log?

Type in a terminal window: PECAN_PERSIST=1 PECAN_DEBUG=all 
/usr/lib/telepathy/tp-msn-pecan
Open Empathy

Some informations about tp-msn-pecan (telepathy branch in my git repo):
 - it's a telepathy-haze fork
 - file transfer is not supported (kenshin is working on it in telepathy-haze)
 - you cannot block/unblock contacts (Empathy does not support it)
 - you cannot set the hidden (invisible) status (due to a telepathy bug)
```

Original comment by `devid...@gmail.com` on 30 Aug 2009 at 11:11

**Comment by GoogleCodeExporter** [comment]
```
Hi, I went on Empathy today but msn-pecan's settings page doesn't appear when I 
click
on it.  Nothing appeared in the debug log while I was in Empathy.  Look's like
Empathy has been updated recently as there are remove icons next to each 
protocol
rather than checkboxes, which seems a little counter-intuitive but that's a 
separate
issue.
```

Original comment by `wyattmei...@gmail.com` on 2 Sep 2009 at 6:30

**Comment by GoogleCodeExporter** [comment]
```
OK, updated msn-pecan and also re-added to Empathy, seems to be working now but 
there
are still issues with the settings page not appearing.
```

Original comment by `wyattmei...@gmail.com` on 2 Sep 2009 at 6:36

**Comment by GoogleCodeExporter** [comment]
```
@wyattmeister: can you please post a screenshot of the missing settings?
```

Original comment by `devid...@gmail.com` on 2 Sep 2009 at 8:51

**Comment by GoogleCodeExporter** [comment]
```
Here you go.  The settings page appears when msn-pecan is first added but after 
that
it doesn't appear when you click back on it.  I'm using the latest version of 
Empathy
from the Telepathy PPA.
```

Original comment by `wyattmei...@gmail.com` on 2 Sep 2009 at 10:19

Attachments:
- [Screenshot-Accounts.png](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-98/comment-25/Screenshot-Accounts.png)

**Comment by GoogleCodeExporter** [comment]
```
@wyattmeister: is MSN affected too?
```

Original comment by `devid...@gmail.com` on 2 Sep 2009 at 10:26

**Comment by GoogleCodeExporter** [comment]
```
Nope, just msn-pecan.
```

Original comment by `wyattmei...@gmail.com` on 2 Sep 2009 at 5:09

**Comment by GoogleCodeExporter** [comment]
```
Mhhh... that's strange, tp-msn-pecan is based on the same source code of 
telepathy-haze and the options are 
the same...
```

Original comment by `devid...@gmail.com` on 2 Sep 2009 at 5:57

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Are you still working on this? Would you be kind enough to provide a compiled 
version
of tp-msn-pecan?
```

Original comment by `ricardof...@gmail.com` on 30 Apr 2010 at 1:58

**Comment by GoogleCodeExporter** [comment]
```
@ricardofcferreira there are two ways of having telepathy support; natively or 
through 
libpurple.

I can provide sourcecode to have tp support through libpurple, but the native 
support 
is still work in progress.
```

Original comment by `felipe.contreras` on 30 Apr 2010 at 9:15

**Comment by GoogleCodeExporter** [comment]
```
Quick poll. There are three options:

1) Make sure msn-pecan works nicely telepathy-haze

This is mostly done, but at least on Fedora14 msn-pecan appears without an 
icon, so some small improvements would be needed.

2) Distribute a bulky telepathy-msn-pecan

This would include internally telepathy-haze and libpurple-mini, which is 
exactly what I do for the Nokia N900.

3) A native, slim version of telepathy-msn-pecan

This is is work in progress, and will take a long time I think.

I ask because I can do 1) and 2) easily, so we can aim for 0.2, but 3) would 
take more time, probably 1.0.

Thoughts?
```

Original comment by `felipe.contreras` on 16 May 2011 at 11:48

**Comment by GoogleCodeExporter** [comment]
```
I would prefer 2, we can use that work as base for 3 (remove unused purple/haze 
stuff, cleanup the code, merge useful code in pecan in order to avoid depending 
on purple, etc.)
```

Original comment by `devid...@gmail.com` on 16 May 2011 at 1:31

**Comment by GoogleCodeExporter** [comment]
```
Maybe 2) would be feasible for that, but it would be a bit difficult to 
synchronize msn-pecan and telepathy-msn-pecan that way. I'm not sure, but we 
can start with 2) and see where things go from there.
```

Original comment by `felipe.contreras` on 16 May 2011 at 2:02
